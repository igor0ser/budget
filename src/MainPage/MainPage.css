.MainPage__header {
  display: flex;
  height: 20vh;
  justify-content: space-between;
  align-items: center;
  padding: 0 30px;
  border-bottom: 2px solid purple;
  text-transform: uppercase;
  font-size: 4vh;
}

.MainPage__logo {
  height: 70%;
  width: auto;
}

.MainPage__btnWrapper {
  display: flex;
  flex-direction: column;
  height: 80vh;
  justify-content: space-around;
  align-items: center;
}

.MainPage .btn {
  width: 80vw;
  height: 18vh;
  font-size: 4vh;
  border-radius: 3vw;
}