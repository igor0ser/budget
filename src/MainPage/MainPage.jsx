import React from 'react';
import PropTypes from 'prop-types';
import logo from './images/logo.png';
import './MainPage.css';

const MainPage = ({ goToManageBudgets, goToAddExpense }) => (
  <div className="MainPage">
    <header className="MainPage__header">
      <img src={logo} alt="logo" className="MainPage__logo" />
      your budget
    </header>
    <div className="MainPage__btnWrapper">
      <button
        className="btn MainPage__btn"
        onClick={goToManageBudgets}
      >
        Manage Budgets
      </button>
      <button
        className="btn MainPage__btn"
        onClick={goToAddExpense}
      >
        Add Expense
      </button>
    </div>
  </div>
);

MainPage.propTypes = {
  goToManageBudgets: PropTypes.func.isRequired,
  goToAddExpense: PropTypes.func.isRequired,
};

export default MainPage;
