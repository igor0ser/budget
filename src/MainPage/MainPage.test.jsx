import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import MainPage from './MainPage';

describe('MainPage', () => {
  let comp;
  let goToManageBudgets;
  let goToAddExpense;

  beforeEach(() => {
    goToManageBudgets = jest.fn();
    goToAddExpense = jest.fn();

    comp = shallow(
      <MainPage
        goToManageBudgets={goToManageBudgets}
        goToAddExpense={goToAddExpense}
      />,
    );
  });

  afterEach(() => {
    comp.unmount();
  });

  it('has 2 buttons', () => {
    expect(comp.find('button.MainPage__btn')).toHaveLength(2);
    expect(toJson(comp)).toMatchSnapshot();
  });

  it('invokes goToManageBudgets on first button click', () => {
    comp.find('button').first().simulate('click');
    expect(goToManageBudgets.mock.calls).toHaveLength(1);
  });

  it('invokes goToAddExpense on second button click', () => {
    comp.find('button').last().simulate('click');
    expect(goToAddExpense.mock.calls).toHaveLength(1);
  });
});
