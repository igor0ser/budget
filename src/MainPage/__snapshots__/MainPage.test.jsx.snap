// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPage has 2 buttons 1`] = `
<div
  className="MainPage"
>
  <header
    className="MainPage__header"
  >
    <img
      alt=""
      className="MainPage__logo"
      src=""
    />
    Budget
  </header>
  <button
    className="btn MainPage__btn"
    onClick={[Function]}
  >
    Manage Budgets
  </button>
  <button
    className="btn MainPage__btn"
    onClick={[Function]}
  >
    Add Expense
  </button>
</div>
`;
